CesiumUtility/src/Uri.cpp

```diff
@@ -10,20 +10,35 @@
 #include <vector>
 
 namespace CesiumUtility {
+const char* HTTPS_PREFIX = "https";
+const char* HTTPS_PREFIX_COLON = "https:";
+
+std::string cesiumConformUrl(const std::string& url) {
+  if (url.rfind("//", 0) == 0) {
+    return std::string(HTTPS_PREFIX_COLON).append(url);
+  } else if (url.find("://", 0) == 0) {
+    return std::string(HTTPS_PREFIX).append(url);
+  }
+  return url;
+}
+
 std::string Uri::resolve(
     const std::string& base,
     const std::string& relative,
     bool useBaseQuery) {
+  const std::string conformedBase = cesiumConformUrl(base);
```

Comment:
I noticed we assume the current protocol is HTTPS in `cesiumConformUrl`.

Probably safe, but another option would be to have the caller pass in the current protocol (maybe as an optional 4th arg?) 

Then we can support HTTP too, avoid any complicated 'what protocol should we default to' logic, and let the caller make its own assumptions.